@per1234 ,

> Please always thoroughly test your code before submitting a pull request.
> 
> You have introduced a compiler warning:
> 
> ```
> MKRTHERM.cpp:111:1: warning: control reaches end of non-void function [-Wreturn-type]
> ```
> 
> My instinct is to add `return NAN;` to the end of the function, but I took a look at the Arduino_HTS221 library and it just defaults to returning the Celsius value:
> https://github.com/arduino-libraries/Arduino_HTS221/blob/0f309b2633f89bd44a1779d43b39448c5665527b/src/HTS.cpp#L89-L93
> I'll leave it up to you to decide how to fix the warning.

I have made the function default to the celsius setting similar to the HTS library.

> Please add the `FAHRENHEIT`, `CELCIUS`, and `KELVIN` keywords to keywords.txt, as was done in the Arduino_HTS221 library:
> https://github.com/arduino-libraries/Arduino_HTS221/blob/0f309b2633f89bd44a1779d43b39448c5665527b/keywords.txt#L25-L26
> Be sure to use a true tab field separator, not spaces.
> https://arduino.github.io/arduino-cli/library-specification/#keywordstxt-format

I have added the keywords. I apologise for not adding them before since I was not aware.
Please recheck the tab setting (I added a single tab but I am not sure about my editor's autoindent.)




also, @per1234 , I am applying for GSoC this year and have written a draft proposal for the same, since you seem to be a very active member of this community, I would be very grateful if you could  please review my proposal and suggest changes

--- a/keywords.txt
+++ b/keywords.txt
@@ -1,6 +1,6 @@
 #######################################
 # Syntax Coloring Map For Arduino_MKRTHERM
-####################################### 
+#######################################
 # Class
 #######################################
 
@@ -9,8 +9,8 @@ MKRTHERM	KEYWORD1
 THERM	KEYWORD1
 
 #######################################
-# Methods and Functions 
-#######################################	
+# Methods and Functions
+#######################################
 
 begin	KEYWORD2
 end	KEYWORD2
@@ -21,3 +21,7 @@ readReferenceTemperature	KEYWORD2
 #######################################
 # Constants
 #######################################
+
+FAHRENHEIT	LITERAL1
+CELSIUS	LITERAL1
+KELVIN	LITERAL1

